solr/core/src/test/org/apache/solr/cloud/hdfs/HdfsTestUtil.java

```diff
@@ -62,73 +63,88 @@
   public static MiniDFSCluster setupClass(String dir) throws Exception {
     return setupClass(dir, true, true);
   }
-  
+
   public static MiniDFSCluster setupClass(String dir, boolean haTesting) throws Exception {
     return setupClass(dir, haTesting, true);
   }
-  
+
+  /**
+   * Checks that commons-lang3 FastDateFormat works with configured locale
+   */
+  @SuppressForbidden(reason="Call FastDateFormat.format same way Hadoop calls it")
+  private static void checkFastDateFormat() {
+    try {
+      FastDateFormat.getInstance().format(System.currentTimeMillis());
+    } catch (ArrayIndexOutOfBoundsException e) {
+      LuceneTestCase.assumeNoException("commons-lang3 FastDateFormat doesn't work with " +
+          Locale.getDefault().toLanguageTag(), e);
+    }
+  }
+
+  /**
+   * Hadoop fails to generate locale agnostic ids - Checks that generated string matches
+   */
+  private static void checkGeneratedIdMatches() {
```

Comment:
Hadoop generates filename ids with `String.format` without any Locale set and then uses regex to try to match it. This fails on locales `th-TH-u-nu-thai-x-lvariant-TH` and `hi-IN`. This check matches what Hadoop is doing to make sure there is a match.